### PR TITLE
feat(css_parser): CSS Parser pseudo class selector #268

### PR DESCRIPTION
**Pull request by denbezrukov**
## Summary

https://developer.mozilla.org/en-US/docs/Web/CSS/Pseudo-classes

Pseudo class selector syntax:

```
AnyCssPseudoClass =
	CssPseudoClassIdentifier                     // :first-of-type {}
	| CssPseudoClassFunctionIdentifier           // :dir(ltr) {}
	| CssPseudoClassFunctionSelector             // :global(.class div) {}
	| CssPseudoClassFunctionSelectorList         // :not(div, .fancy) {}
	| CssPseudoClassFunctionCompoundSelector     // :host(span:focus) {}
	| CssPseudoClassFunctionCompoundSelectorList // :-webkit-any(i,p,:link,span:focus) {}
	| CssPseudoClassFunctionRelativeSelectorList // :has(> img, +dt) {}
```

## Test Plan

`cargo test -p biome_css_parser`

### CODE REVIEW

**Comment by ematipico** [comment]
Can you give us more information about your decisions to breakdown the pseudo selector in multiple nodes?

From the way I see it `:global(.class div)` and `host(span:focus)` could be the same node:


```
CssPseudoClassFunctionSelectorList =
 	name: CssIdentifier
	'('
	list: CssSelectorList
	')

CssSelectorList = ...
```

Where `CssSelectorList` will include all these nodes you identified: identifier, selector, compound, etc.

File: crates/biome_css_parser/src/syntax/selector/attribute.rs [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+use crate::syntax::{is_at_identifier, parse_css_string, parse_regular_identifier};
+use biome_css_syntax::CssSyntaxKind::*;
+use biome_css_syntax::{CssSyntaxKind, T};
+use biome_parser::diagnostic::expected_token;
+use biome_parser::parse_recovery::ParseRecovery;
+use biome_parser::prelude::ParsedSyntax;
+use biome_parser::prelude::ParsedSyntax::{Absent, Present};
+use biome_parser::{token_set, Parser, TokenSet};
+
+const ATTRIBUTE_SELECTOR_RECOVERY_SET: TokenSet<CssSyntaxKind> = token_set![T![')'], T!['{']];
```
Comment (ematipico) [inline crates/biome_css_parser/src/syntax/selector/attribute.rs:15]: Can you add a small comment that shows why/when these tokens do a recovery? I can guess two cases:

```css
:hover) {
/*    ^ recovers here */
:hover {
/*     ^ recovers here */
```

**Comment by denbezrukov** [comment]
@ematipico 
Current implementation follows the CSS specifications. 
For instance, `:host` only accepts a `compound-selector` ([CSS Scoping 1](https://drafts.csswg.org/css-scoping/#selectordef-host-function), [MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/:host_function#syntax)), and `:not` only takes a `<complex-selector-list>` ([MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/:not#syntax)).

**Review by ematipico — COMMENTED** [review]
Overall I think it's good that we are re-using parsing logics. I left a question (in the previous comment) about giving us more information about why you brokedown the new sytanx in difficult nodes.

I noticed you added a many "OK" tests, which is good, although you haven't error cases. Could you add some of them, especially cases that demonstrate the recovery of the new nodes. Without them, I find very difficult to give some feedback in this PR.

**Review by ematipico — COMMENTED** [review]
Overall I think it's good that we are re-using parsing logics. I left a question (in the previous comment) about giving us more information about why you brokedown the new sytanx in difficult nodes.

I noticed you added a many "OK" tests, which is good, although you haven't added error cases. Could you add some of them, especially cases that demonstrate the recovery of the new nodes. Without them, I find very difficult to give some feedback in this PR.

File: crates/biome_css_parser/src/syntax/selector/attribute.rs (hunk 1 above)
Comment (denbezrukov) [reply crates/biome_css_parser/src/syntax/selector/attribute.rs:15]: Sure,
I was wondering if we could do it in another PR? 
I'd like to merge this one and the next with pseudo class nth selector.

After I'm going to add error cases for every pseudo class selector.
What do you think?

Comment (ematipico) [reply crates/biome_css_parser/src/syntax/selector/attribute.rs:15]: Sounds good 😊

**Pull request merged**